File: 0121. Best Time to Buy and Sell Stock/0121.java
class Solution {
    public int maxProfit(int[] prices) {
        // find the difference between the smallest and the largest, but it should check from left to right
        // so, it should be in increasing order
        if (prices == null || prices.length == 0) return 0;
        int lowest, highest;
        lowest = highest = prices[0];
        int difference=0;
        for(int i=0; i<prices.length; i++){
            if(prices[i]<lowest){
                lowest=prices[i];
                highest=prices[i];
            }
            if(prices[i]>highest){
                highest=prices[i];
            }
            if(difference<highest-lowest) difference = highest-lowest;
        }
        return difference;
    }
}

File: 0121. Best Time to Buy and Sell Stock/0121.cpp
class Solution {
    public:
        int maxProfit(vector<int>& prices) {
            // find the difference between the smallest and the largest, but it should check from left to right
            int min = prices[0];
            int difference = 0;
            for (int i=1; i<prices.size(); i++){
                // if(difference < prices[i] - min) difference = prices[i] - min;
                difference = max(difference, prices[i]-min);
                if(min > prices[i]) min=prices[i];
            }
            return difference;
        }
    };